Realm/RLMConfiguration.m

```diff
@@ -0,0 +1,159 @@
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMConfiguration_Private.h"
+
+static NSString * const c_RLMConfigurationProperties[] = {
+    @"path",
+    @"inMemoryIdentifier",
+    @"encryptionKey",
+    @"readOnly",
+    @"schemaVersion",
+    @"migrationBlock",
+    @"dynamic",
+    @"customSchema",
+};
+static const NSUInteger c_RLMConfigurationPropertiesCount = sizeof(c_RLMConfigurationProperties) / sizeof(NSString *);
+
+@interface RLMConfiguration ()
+
+@property (nonatomic, copy, readwrite) NSString *path;
+@property (nonatomic, copy, readwrite) NSString *inMemoryIdentifier;
+@property (nonatomic, copy, readwrite) NSData *encryptionKey;
+@property (nonatomic, readwrite)       BOOL    readOnly;
+@property (nonatomic, copy, readwrite) NSString *fileProtectionAttributes;
+
+@property (nonatomic, readwrite)       NSUInteger schemaVersion;
+@property (nonatomic, copy, readwrite) RLMMigrationBlock migrationBlock;
+
+@end
+
+@interface RLMConfiguration ()<RLMConfigurator>
+
+@end
+
+@implementation RLMConfiguration
+
+RLMConfiguration *s_defaultConfiguration;
+static NSString * const c_defaultRealmFileName = @"default.realm";
+
++ (NSString *)defaultRealmPath
+{
+    static NSString *defaultRealmPath;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        defaultRealmPath = [[self class] writeablePathForFile:c_defaultRealmFileName];
+    });
+    return defaultRealmPath;
+}
+
++ (NSString *)writeablePathForFile:(NSString*)fileName
+{
+#if TARGET_OS_IPHONE
+    // On iOS the Documents directory isn't user-visible, so put files there
+    NSString *path = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES)[0];
+#else
+    // On OS X it is, so put files in Application Support. If we aren't running
+    // in a sandbox, put it in a subdirectory based on the bundle identifier
+    // to avoid accidentally sharing files between applications
+    NSString *path = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES)[0];
+    if (![[NSProcessInfo processInfo] environment][@"APP_SANDBOX_CONTAINER_ID"]) {
+        NSString *identifier = [[NSBundle mainBundle] bundleIdentifier];
+        if ([identifier length] == 0) {
+            identifier = [[[NSBundle mainBundle] executablePath] lastPathComponent];
+        }
+        path = [path stringByAppendingPathComponent:identifier];
+
+        // create directory
+        [[NSFileManager defaultManager] createDirectoryAtPath:path
+                                  withIntermediateDirectories:YES
+                                                   attributes:nil
+                                                        error:nil];
+    }
+#endif
+    return [path stringByAppendingPathComponent:fileName];
+}
+
++ (instancetype)defaultConfiguration {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        if (!s_defaultConfiguration) {
```

Comment:
why? the dispatch_once block, after it's run the first time, is basically only a conditional